installscripts/wizard/validate_tags.py

```diff
@@ -0,0 +1,83 @@
+    - the 'Key' Values should be unique
+    - the 'Key' and 'Value' Values must not start with 'aws'
+    - the 'Key' and 'Value' should be case sensitive
+    - the 'Key' and 'Value' should be alphanumeric plus the \
+      following special characters: + - = . _ : / @. with spaces
+    """
+
+    # helper method to validate Keys and Values
+    def validate_string(val, s_type, s_max_length):
+        if type(val) != str:
+            raise ValueError("Non string " + s_type + " found:" + str(val))
+        str_val = val.strip()
+        if not len(str_val):
+            raise ValueError("Empty " + s_type + " found.")
+        if len(str_val) >= s_max_length:
+            raise ValueError("Too long " + s_type + " found: " + str_val)
+        if str_val.lower().startswith("aws"):
+            raise ValueError(s_type + " starting with aws found: " + str_val)
+        return str_val
+
+    # helper method to validate spedical characters in Keys or Values
+    def validate_special_characters(item):
+        if not re.match("^[a-zA-Z0-9\s\+\-\=\.\_\:\/\@\.]*$", item):
+            raise ValueError("tag encountered with special character: " + item)
+
+    # helper method to validate case sensitive
+    def validate_case_sensitive(item):
+        to_check_str = item
+        for character in ["+", "-", "=", ".", "_", ":", "/", "@", ".", " "]:
+            to_check_str = to_check_str.replace(character, "")
+        if not to_check_str.islower() and not to_check_str.isupper() and to_check_str.isalnum():
+            raise ValueError("Tag Keys and Values are case sensitive: " + item)
+
+    if type(replication_tags) != list:
+        raise ValueError("replication_tags should be a list")
+
+    reserved_tags = ["name", "application", "jazzinstance", "environment", "exempt", "owner"]
+    unique_tags = []
+    new_replication_tags = []
+    tag_for_resource = '{'
+    for tag in replication_tags:
+
+        if not (type(tag) == dict and len(tag) == 2 and
+                'Key' in tag and 'Value' in tag):
+            raise ValueError("tags should be dicts with Keys 'Key' and 'Value'")
+
+        Key = validate_string(tag['Key'], 'Key', 127)
+        Value = validate_string(tag['Value'], 'Value', 255)
+
+        validate_special_characters(Key)
+        validate_special_characters(Value)
+        validate_case_sensitive(Key)
+        validate_case_sensitive(Value)
+
+        if Key in reserved_tags:
+            raise ValueError("tag encountered with reserved Key: " + Key)
+
+        if Key in unique_tags:
+            raise ValueError("tag encountered with duplicate Key: " + Key)
+
+        unique_tags.append(Key)
+        new_replication_tags.append({"Key": Key, "Value": Value})
+        tag_for_resource += Key+'="'+Value + '", '
+        if len(new_replication_tags) > 49:
+            raise ValueError("More than 50 tags in replication settings.")
+
+    return new_replication_tags, tag_for_resource
+
+
+def prepare_tags(input_tags_param):
```

Comment:
Can you create a handful of Python unit tests for this module? It's standalone, complex enough to need validation, and is generally a good target for unit tests.